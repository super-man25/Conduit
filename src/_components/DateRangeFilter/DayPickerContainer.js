import styled, { css } from 'styled-components';
import { cssConstants, shadows } from '_constants';

export const DayPickerContainer = styled.div`
  display: flex;
  align-items: center;
  align-content: space-between;
  flex-direction: column;
  border: 1px solid ${cssConstants.PRIMARY_LIGHT_BLUE};
  border-left: none;
  background: ${cssConstants.PRIMARY_WHITE};
  opacity: 0;
  visibility: hidden;
  transition: all 0.1s ease-in-out;

  ${(props) =>
    props.show &&
    css`
      opacity: 1;
      transform: translateY(0);
      visibility: visible;
      box-shadow: ${shadows.SMALL};
      border-radius: 0 4px 4px 0;
    `};
`;
DayPickerContainer.displayName = 'DropdownMenu';
