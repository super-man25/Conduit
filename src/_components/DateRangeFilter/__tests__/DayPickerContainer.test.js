import React from 'react';
import renderer from 'react-test-renderer';
import { DayPickerContainer } from '../DayPickerContainer';

it('renders correctly when not showing', () => {
  const tree = renderer.create(<DayPickerContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when showing', () => {
  const tree = renderer.create(<DayPickerContainer show />).toJSON();
  expect(tree).toMatchSnapshot();
});
