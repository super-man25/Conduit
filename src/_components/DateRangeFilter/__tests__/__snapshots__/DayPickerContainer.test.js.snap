// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not showing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #0A169A;
  border-left: none;
  background: #FFFFFF;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
}

<div
  className="c0"
/>
`;

exports[`renders correctly when showing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #0A169A;
  border-left: none;
  background: #FFFFFF;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  opacity: 1;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  visibility: visible;
  box-shadow: 0 1px 5px rgba(0,0,0,0.1);
  border-radius: 0 4px 4px 0;
}

<div
  className="c0"
/>
`;
